### PR TITLE
Add tests for weird characters

### PR DESCRIPTION
**Pull request by carlossg**
Showing that jclouds can not handle s3 keys with ampersand '&'

### CODE REVIEW

**Comment by jglick** [comment]
Seeing a test failure

```
java.lang.NoClassDefFoundError: hudson/plugins/sshslaves/SSHLauncher
	at io.jenkins.plugins.artifact_manager_s3.JCloudsArtifactManagerTest.smokes(JCloudsArtifactManagerTest.java:110)
```

which does not make much sense—wires getting crossed with #7?

File: src/test/java/io/jenkins/plugins/artifact_manager_s3/JCloudsBlobStoreTest.java [hunk 1]
```diff
@@ -24,31 +24,33 @@
 
 package io.jenkins.plugins.artifact_manager_s3;
 
+import static org.hamcrest.Matchers.*;
```
Comment (jglick) [inline src/test/java/io/jenkins/plugins/artifact_manager_s3/JCloudsBlobStoreTest.java]: Urgh, we need to adopt a consistent import format guideline. I generally just put them all in one block, sorted, as it is simplest and least vulnerable to merge conflicts.

Comment (carlossg) [reply src/test/java/io/jenkins/plugins/artifact_manager_s3/JCloudsBlobStoreTest.java]: I use the default Eclipse Java guidelines, with longer lines IIRC

**Comment by carlossg** [comment]
Filed the bug as https://issues.jenkins-ci.org/browse/JENKINS-50591 and https://issues.apache.org/jira/browse/JCLOUDS-1401
Anyway the chances of hitting a path with `&` are pretty low, just leaving it as a known issue

**Pull request merged**